internal/runtime_reflect/slice.go

```diff
@@ -39,6 +39,6 @@ func CopySlice(elemType unsafe.Pointer, dst, src Slice) int {
 //go:linkname newarray runtime.newarray
 func newarray(t unsafe.Pointer, n int) unsafe.Pointer
 
-//go:linkname typedslicecopy runtime.typedslicecopy
+//go:linkname typedslicecopy reflect.typedslicecopy
```

Comment:
The current version of the code seems to build and pass tests with Go 1.15 to 1.17, do you mind sharing the error you were seeing, and the version of Go you were using?

Reply:
using go1.17.1 and the error message is following

```
[2021-11-22 22:38:15] [ERROR]: Protocol -> panic on decoder trpc.group_pro.synclogic.SyncLogic.PushFirstView : runtime error: invalid memory address or nil pointer deref
erence
goroutine 119 [running]:
runtime/debug.Stack()
        C:/Users/wdvxdr/sdk/go1.17.1/src/runtime/debug/stack.go:24 +0x65
github.com/Mrs4s/MiraiGo/client.(*QQClient).netLoop.func1.1()
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/network.go:323 +0x5d
panic({0x169d820, 0x1eff220})
        C:/Users/wdvxdr/sdk/go1.17.1/src/runtime/panic.go:1038 +0x215
github.com/segmentio/encoding/internal/runtime_reflect.CopySlice(...)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/internal/runtime_reflect/slice.go:36
github.com/segmentio/encoding/proto.growSlice({0x18f4370, 0x171b940}, 0xc0002e8268)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/slice.go:139 +0x89
github.com/segmentio/encoding/proto.sliceDecodeFuncOf.func1({0xc00030b178, 0x19, 0xe77}, 0xc0002e8268, 0x0)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/slice.go:108 +0x6d
github.com/segmentio/encoding/proto.structDecodeFuncOf.func1({0xc00030b00f, 0x1ef, 0xfe0}, 0xc0002e81e0, 0x0)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/struct.go:402 +0x64e
github.com/segmentio/encoding/proto.pointerDecodeFuncOf.func1({0xc00030b00f, 0x1ef, 0xfe0}, 0xc0003cc230, 0xc00030c548)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/pointer.go:50 +0xa6
github.com/segmentio/encoding/proto.sliceDecodeFuncOf.func1({0xc00030b00f, 0x1ef, 0xfe0}, 0xc0003cc210, 0xc000587820)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/slice.go:111 +0xc4
github.com/segmentio/encoding/proto.structDecodeFuncOf.func1({0xc00030b003, 0xfec, 0xfec}, 0xc0003cc1e0, 0x0)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/struct.go:402 +0x64e
github.com/segmentio/encoding/proto.pointerDecodeFuncOf.func1({0xc00030b003, 0xfec, 0xfec}, 0xc0004cc178, 0x2)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/pointer.go:50 +0xa6
github.com/segmentio/encoding/proto.structDecodeFuncOf.func1({0xc00030b000, 0xfef, 0xfef}, 0xc0004cc150, 0x70)
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/struct.go:402 +0x64e
github.com/segmentio/encoding/proto.Unmarshal({0xc00030b000, 0xfef, 0xfef}, {0x16e2c40, 0xc0004cc150})
        C:/Users/wdvxdr/go/pkg/mod/github.com/segmentio/encoding@v0.3.2/proto/proto.go:49 +0x1d7
github.com/Mrs4s/MiraiGo/internal/proto.Unmarshal(...)
        C:/Users/wdvxdr/Documents/Project/MiraiGo/internal/proto/wrapper.go:12
github.com/Mrs4s/MiraiGo/client.(*QQClient).unpackOIDBPackage(0xc000444480, {0xc000553000, 0xffb, 0xffb}, {0x16e2c40, 0xc0004cc150})
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/global.go:674 +0x154
github.com/Mrs4s/MiraiGo/client.(*GuildService).FetchChannelList(0xc0004161e0, 0x2c)
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/guild.go:501 +0x313
github.com/Mrs4s/MiraiGo/client.decodeGuildPushFirstView(0xc000444480, 0x11c, {0xc00046e5a0, 0x11c, 0x11c})
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/guild.go:630 +0x387
github.com/Mrs4s/MiraiGo/client.(*QQClient).netLoop.func1(0xc0004173b0)
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/network.go:333 +0x1cb
created by github.com/Mrs4s/MiraiGo/client.(*QQClient).netLoop
        C:/Users/wdvxdr/Documents/Project/MiraiGo/client/network.go:320 +0x652
```